src/public/js/rs.js

```diff
@@ -45,13 +45,214 @@ var RS = (function() {
       }
     }
     if (errors) {
-      // TODO ERROR CORRECTION
-      throw new Error('Error correction not implemented yet');
+      // Perform B-M decoding
+      var syndromes = computeSyndromePoly(receivedMsg, n, k);
+      var bm = berlekampMassey(syndromes, n, k);
+      var errorLocations = chienSearch(bm.errorLocationPoly);
+      var errorMagnitudes = forneysFormula(errorLocations,
+                                           bm.errorMagnitudePoly);
+      var repairedMsg = repair(receivedMsg, errorLocations, errorMagnitudes);
+
+      // Check that the repaired message is in fact divisible by genPoly
+      var repairedRemainder = UTIL.polynomialDiv(repairedMsg, genPoly, n, k);
+      var repairedErrors = false;
+      for (var i = 0; i < n - k; i++) {
+        if (repairedRemainder[i] != 0) {
```

Comment:
I think if there are too many errors to decode we may still get a 0 remainder. I am not sure I need to think further on this. It may be that we get 0 remainder it is just the wrong polynomial that is evenly divided by G(x).